trade-finance-manager-api/src/graphql/resolvers/query-facilities.js

```diff
@@ -8,17 +10,42 @@ exports.queryAllFacilities = async (queryParams) => {
   const facilities = [];
   // eslint-disable-next-line no-restricted-syntax
   for (const item of rawFacilities) {
-    const facility = item.tfmFacilities;
-    const defaultCoverEndDate = item.tfmFacilities.coverEndDate;
+    const { tfmFacilities: facility } = item;
+    const { facilityId } = facility;
+
+    // checks for amended facility value or coverEndDate
+    const latestCompletedAmendmentValue = await api.getLatestCompletedValueAmendment(facilityId);
+    const latestCompletedAmendmentCoverEndDate = await api.getLatestCompletedDateAmendment(facilityId);
+
+    const defaultFacilityValue = facility.value;
+    const defaultCoverEndDate = facility.coverEndDate;
     const formatCoverEndDate = format(new Date(defaultCoverEndDate), 'dd LLL yyyy'); // 11 Aug 2021
 
-    facility.coverEndDate = item.tfmFacilities.coverEndDate ? formatCoverEndDate : '';
-    const defaultFacilityValue = item.tfmFacilities.value;
+    // sets the values based on original facility value
+    let currencyAndValue = `${facility.currency} ${commaNumber(defaultFacilityValue)}`;
+    let facilityValue = parseInt(facility.value, 10);
+    let facilityCoverEndDate = facility.coverEndDate ? formatCoverEndDate : '';
+    let coverEndDateEpoch = facility.coverEndDate ? getUnixTime(new Date(facility.coverEndDate)) : '';
+
+    // if amendment, then willset relevant values based on amendment
+    if (latestCompletedAmendmentValue?.value) {
+      const { value, currency } = latestCompletedAmendmentValue;
+      currencyAndValue = `${currency} ${commaNumber(value)}`;
+      facilityValue = parseInt(value, 10);
+    }
+
+    if (latestCompletedAmendmentCoverEndDate?.coverEndDate) {
+      const { coverEndDate } = latestCompletedAmendmentCoverEndDate;
+      facilityCoverEndDate = format(new Date(coverEndDate * 1000), 'dd LLL yyyy');
```

Comment:
We know why we are adding `1000` can you please add a comment line explaining our decision.

Reply:
done